test_start_span_custom_start_time fails randomly

```
=================================== FAILURES ===================================
_________________ TestTracer.test_start_span_custom_start_time _________________

self = <tests.opentracer.test_tracer.TestTracer object at 0x7fe6bdcb8a58>
ot_tracer = <ddtrace.opentracer.tracer.Tracer object at 0x7fe6be32ccc0>

    def test_start_span_custom_start_time(self, ot_tracer):
        """Start a span with a custom start time."""
        import time
    
        t = time.time() + 0.002
        with ot_tracer.start_span('myop', start_time=t) as span:
            time.sleep(0.005)
    
        # it should be certain that the span duration is strictly less than
        # the amount of time we sleep for
>       assert span._dd_span.duration < 0.005
E       assert 0.007131099700927734 < 0.005
E        +  where 0.007131099700927734 = <Span(id=760474509597788229,trace_id=3017630132355656079,parent_id=None,name=myop)>.duration
E        +    where <Span(id=760474509597788229,trace_id=3017630132355656079,parent_id=None,name=myop)> = <ddtrace.opentracer.span.Span object at 0x7fe6be32cf60>._dd_span

tests/opentracer/test_tracer.py:136: AssertionError
===================== 1 failed, 63 passed in 0.61 seconds ======================
```
https://circleci.com/gh/DataDog/dd-trace-py/118474

--- a/tests/opentracer/test_tracer.py
+++ b/tests/opentracer/test_tracer.py
@@ -16,6 +16,7 @@ from ddtrace.opentracer.span_context import SpanContext
 from ddtrace.propagation.http import HTTP_HEADER_TRACE_ID
 from ddtrace.settings import ConfigException
 
+import mock
 import pytest
 
 
@@ -125,13 +126,14 @@ class TestTracer(object):
 
     def test_start_span_custom_start_time(self, ot_tracer):
         """Start a span with a custom start time."""
-        t = time.time() + 0.002
-        with ot_tracer.start_span('myop', start_time=t) as span:
-            time.sleep(0.005)
+        t = 100
+        with mock.patch('time.time') as time:
+            time.return_value = 102
+            with ot_tracer.start_span('myop', start_time=t) as span:
+                pass
 
-        # it should be certain that the span duration is strictly less than
-        # the amount of time we sleep for
-        assert span._dd_span.duration < 0.005
+        assert span._dd_span.start == t
+        assert span._dd_span.duration == 2
 
     def test_start_span_with_spancontext(self, ot_tracer, writer):
         """Start and finish a span using a span context as the child_of

